runtime/vam/op/summarize/summarize.go

```diff
@@ -0,0 +1,110 @@
+package summarize
+
+import (
+	zed "github.com/brimdata/super"
+	"github.com/brimdata/super/pkg/field"
+	"github.com/brimdata/super/runtime/vam/expr"
+	"github.com/brimdata/super/vector"
+)
+
+type Summarize struct {
+	parent vector.Puller
+	zctx   *zed.Context
+	// XX Abstract this runtime into a generic table computation.
+	// Then the generic interface can execute fast paths for simple scenarios.
+	aggs      []*expr.Aggregator
+	aggNames  field.List
+	keyExprs  []expr.Evaluator
+	keyNames  field.List
+	typeTable *zed.TypeVectorTable
+	builder   *vector.RecordBuilder
+
+	types   []zed.Type
+	tables  map[int]aggTable
+	results []aggTable
+}
+
+func New(parent vector.Puller, zctx *zed.Context, aggNames field.List, aggs []*expr.Aggregator, keyNames []field.Path, keyExprs []expr.Evaluator) (*Summarize, error) {
+	builder, err := vector.NewRecordBuilder(zctx, append(keyNames, aggNames...))
+	if err != nil {
+		return nil, err
+	}
+	return &Summarize{
+		parent:    parent,
+		aggs:      aggs,
+		keyExprs:  keyExprs,
+		tables:    make(map[int]aggTable),
+		typeTable: zed.NewTypeVectorTable(),
+		types:     make([]zed.Type, len(keyExprs)),
+		builder:   builder,
+	}, nil
+}
+
+func (s *Summarize) Pull(done bool) (vector.Any, error) {
+	if done {
+		_, err := s.parent.Pull(done)
+		return nil, err
+	}
+	if s.results != nil {
+		return s.next(), nil
+	}
+	for {
+		//XXX check context Done
+		vec, err := s.parent.Pull(false)
+		if err != nil {
+			return nil, err
+		}
+		if vec == nil {
+			for _, t := range s.tables {
+				s.results = append(s.results, t)
+			}
+			s.tables = nil
+			return s.next(), nil
+		}
+		// XXX Eval of keys and vals can be done in parallel.
```

Comment:
I think this was my comment but not clear we want to parallelize here if we want to keep the morsel-based scheduling philosophy.  Would be easy to measure both paths.  Maybe update the comment here?